src/Build/Evaluation/ItemSpec.cs

```diff
@@ -415,6 +420,16 @@ public virtual bool IsMatch(string itemToMatch)
             return FileMatcher.IsMatch(itemToMatch);
         }
 
+        public virtual bool IsMatchNormalized(string normalizedItemToMatch)
+        {
+            return FileMatcher.IsMatchNormalized(normalizedItemToMatch);
+        }
+
+        public virtual IEnumerable<string> GetReferencedItems()
+        {
+            yield return TextFragment;
```

Comment:
Given LINQ is optimized to know about LINQ in chained operations, maybe. Unlike `yield return`, this would have support for speed optimizations like `TryGetCount`.

```suggestion
           return Enumerable.Repeat(TextFragment, 1);
```

Reply:
Wow, I was hoping C# would generate better code for an iterator function yield-returning a single item. I have applied your change. I find it less readable but it wins perf-wise, at least on Core.